src/services/cli-http-client.js

```diff
@@ -64,12 +65,16 @@ class CliRequestClient {
       const b64Auth = Buffer.from(`${opts.username}:${opts.password}`).toString('base64');
       headers.Authorization = `Basic ${b64Auth}`;
     }
-
-    const componentInfo = (headers['User-Agent'] || '').replace(' (', '|').replace(')', '').split('|');
-    componentInfo.push(`${os.platform()} ${os.release()} ${os.arch()}`);
+    const componentInfo = [];
+    componentInfo.push(`(${os.platform()} ${os.arch()})`);
+    const userAgentArr = (headers['User-Agent'] || '').split(' ');
+    componentInfo.push(userAgentArr[0]); // extensions
+    componentInfo.push(userAgentArr[3]); // extensions
```

Comment:
`nit` - Could add some `comments` for what's being processed at each stage

Reply:
Yes, that'd be better. Will do it.

Reply:
Done. Please check..